src/Common/Common.Api/Datasets/DatasetsClient.cs

```diff
@@ -36,7 +36,7 @@ public IEnumerable<Dataset> GetDatasetsAsAdminForWorkspace(Guid workspaceId, str
             return this.Client.Datasets.GetDatasetsAsAdmin(groupId: workspaceId.ToString(), filter: filter, top: top, skip: skip).Value?.Select(x => (Dataset)x);
         }
 
-        public IEnumerable<Datasource> GetDatasources(Guid datasetId, Guid? workspaceId = default)
+        public virtual IEnumerable<Datasource> GetDatasources(Guid datasetId, Guid? workspaceId = default)
```

Comment:
I don't think you should need this virtual keyword applied if you're mocking using the interface.

Reply:
Yeah I will remove, that was the first error I got. Switching to interfaces probably would have prevented me from seeing that